File: 3471-find-the-largest-almost-missing-integer/3471-find-the-largest-almost-missing-integer.c
#include <stdlib.h>

int largestInteger(int* nums, int numsSize, int k) {
    int count[51] = {0};

    for (int i = 0; i < numsSize; i++) {
        count[nums[i]]++;
    }

    // Every element forms its own subarray
    if (k == 1) {
        int ans = -1;

        for (int i = 0; i < numsSize; i++) {
            if (count[nums[i]] == 1) {
                if (nums[i] > ans) {
                    ans = nums[i];
                }
            }
        }

        return ans;
    }

    // Entire array is one subarray
    if (k == numsSize) {
        int ans = nums[0];

        for (int i = 1; i < numsSize; i++) {
            if (nums[i] > ans) {
                ans = nums[i];
            }
        }

        return ans;
    }

    // Only first and last elements can qualify
    int ans = -1;

    if (count[nums[0]] == 1) {
        ans = nums[0];
    }

    if (count[nums[numsSize - 1]] == 1 &&
        nums[numsSize - 1] > ans) {
        ans = nums[numsSize - 1];
    }

    return ans;
}

File: 3471-find-the-largest-almost-missing-integer/3471-find-the-largest-almost-missing-integer.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var largestInteger = function(nums, k) {
    const n = nums.length;

    const count = new Map();

    for (const x of nums) {
        count.set(x, (count.get(x) || 0) + 1);
    }

    // Every element forms its own subarray
    if (k === 1) {
        let ans = -1;

        for (const x of nums) {
            if (count.get(x) === 1) {
                ans = Math.max(ans, x);
            }
        }

        return ans;
    }

    // Entire array is one subarray
    if (k === n) {
        return Math.max(...nums);
    }

    // Only first and last elements can qualify
    let ans = -1;

    if (count.get(nums[0]) === 1) {
        ans = Math.max(ans, nums[0]);
    }

    if (count.get(nums[n - 1]) === 1) {
        ans = Math.max(ans, nums[n - 1]);
    }

    return ans;
};